### PR TITLE
[#PAB-328] enable authentication

### PR DESCRIPTION
**Pull request by cyrusdobbs**
https://trello.com/c/tQ3rDYzh/328-implement-authentication-changes-into-front-end-auth-utils-and-docker-runner

### Change description
Adds the login_required decorator to the download route to enforce auth.

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

**Comment by cyrusdobbs** [comment]
Currently DRAFT. `fsd_utils` changes need to go in first and tests need to be updated / written

File: config.py [hunk 1]
```diff
@@ -13,3 +17,16 @@ class Config(object):
...
     API_HOSTNAME = os.environ.get("API_HOSTNAME")
+
+    USE_LOCAL_DATA = strtobool(os.getenv("USE_LOCAL_DATA", "True"))
+
+    # Funding Service Design Post Award
+    FSD_USER_TOKEN_COOKIE_NAME = "fsd_user_token"
+    AUTHENTICATOR_HOST = os.environ.get("AUTHENTICATOR_HOST", "authenticator")
+    SESSION_COOKIE_DOMAIN = os.environ.get("SESSION_COOKIE_DOMAIN")
+    COOKIE_DOMAIN = os.environ.get("COOKIE_DOMAIN", None)
+
```
Comment (gidsg) [inline config.py:28]: Wonder if we need a SECRET_KEY or at least on deployed environments as per https://flask.palletsprojects.com/en/2.3.x/config/#SECRET_KEY else I think we may get confusing behaviour between environments.

File: config.py [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+from distutils.util import strtobool
+from pathlib import Path
 
 
 class Config(object):
+    FLASK_ROOT = str(Path(__file__).parent)
+
     CONTACT_EMAIL = os.environ.get("CONTACT_EMAIL")
     CONTACT_PHONE = os.environ.get("CONTACT_PHONE")
     DEPARTMENT_NAME = os.environ.get("DEPARTMENT_NAME")
```
Comment (gidsg) [inline config.py:11]: Can we put default values on these? This will prevent us needed to set them in AWS.

File: config.py [hunk 3]
```diff
@@ -13,3 +17,16 @@ class Config(object):
     SESSION_COOKIE_HTTPONLY = True
     SESSION_COOKIE_SECURE = True
     API_HOSTNAME = os.environ.get("API_HOSTNAME")
+
+    USE_LOCAL_DATA = strtobool(os.getenv("USE_LOCAL_DATA", "True"))
```
Comment (gidsg) [inline config.py:21]: Are we using this? Think it could be removed?

File: config.py [hunk 4]
```diff
@@ -13,3 +17,16 @@ class Config(object):
     SESSION_COOKIE_HTTPONLY = True
```
Comment (gidsg) [inline config.py:17]: Are we sure this won't mess up local development by enforcing HTTPS locally?

Comment (cyrusdobbs) [reply config.py:17]: local dev was working with it but is also without and I can't see where it's being used and it isn't present in pre-award-frontend so I'll just remove for now... there were a bunch of env vars i copied across from somewhere when I was first trying to get it set up...

File: config.py [hunk 5]
```diff
@@ -1,15 +1,28 @@
...
+    FSD_USER_TOKEN_COOKIE_NAME = "fsd_user_token"
+    AUTHENTICATOR_HOST = os.environ.get("AUTHENTICATOR_HOST", "authenticator")
+    SESSION_COOKIE_DOMAIN = os.environ.get("SESSION_COOKIE_DOMAIN")
+    COOKIE_DOMAIN = os.environ.get("COOKIE_DOMAIN", None)
+    SECRET_KEY = os.environ.get("SECRET_KEY", "dev")
+
+    # RSA 256 KEYS
+    _test_public_key_path = FLASK_ROOT + "/tests/keys/rsa256/public.pem"
+    with open(_test_public_key_path, mode="rb") as public_key_file:
+        RSA256_PUBLIC_KEY = public_key_file.read()
```
Comment (gidsg) [inline config.py:28]: This needs to be overridable from env

File: config/envs/default.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    CONTACT_EMAIL = os.environ.get("CONTACT_EMAIL", "dev@dev.com")
+    CONTACT_PHONE = os.environ.get("CONTACT_PHONE", "12345678910")
+    DEPARTMENT_NAME = os.environ.get("DEPARTMENT_NAME", "dev-dept-name")
+    DEPARTMENT_URL = os.environ.get("DEPARTMENT_URL", "dev-dept-url")
+    SERVICE_NAME = os.environ.get("SERVICE_NAME", "post-award-frontend")
+    SERVICE_PHASE = os.environ.get("SERVICE_PHASE", "dev-service-phase")
+    SERVICE_URL = os.environ.get("SERVICE_URL", "dev-service-url")
+    SESSION_COOKIE_SECURE = True
+    DATA_STORE_API_HOST = os.environ.get("DATA_STORE_API_HOST")
```
Comment (gidsg) [inline config/envs/default.py:20]: Should we default this to what the docker runner expects?

File: config/envs/default.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    # Funding Service Design Post Award
+    FSD_USER_TOKEN_COOKIE_NAME = "fsd_user_token"
+    AUTHENTICATOR_HOST = os.environ.get("AUTHENTICATOR_HOST", "authenticator")
+    SESSION_COOKIE_DOMAIN = os.environ.get("SESSION_COOKIE_DOMAIN")
```
Comment (gidsg) [inline config/envs/default.py:25]: I have confirmed with @gio-karunakaran this is not used, and was just left over from previous config so can be safely removed.

File: config/envs/default.py (hunk 6 above)
Comment (cyrusdobbs) [reply config/envs/default.py:20]: I think we should just configure in the docker-compose env variables

**Review by gidsg — APPROVED** [review]
LGTM 🔒 👍

**Pull request merged**